### PR TITLE
New working timeline

### PR DESCRIPTION
**Pull request by Codringtoncoding**

### CODE REVIEW

File: src/TimelineID/TimelineId.tsx [hunk 1]
```diff
@@ -1,12 +1,131 @@
...
+                >
+                     <h3 className={styles.TimeLineHeading}>{rover.name}</h3>
+                    <img className={styles.RoverImage} src={rover.imageUrl} alt="rover" />
+                    <p className={styles.RoverInformation}>
+                        {rover.information}
+                    </p>
+                </VerticalTimelineElement>
+            </VerticalTimeline>
+            <Link to="timeline/:roverName" className={styles.ButtonToRoverImage}>{rover.name} Image explorer</Link>
+        </section> //change the line to when image viewer is complete
```
Comment (MichaelWalker) [inline src/TimelineID/TimelineId.tsx:105]: could you delete this comment?

File: src/TimelineID/TimelineId.tsx [hunk 2]
```diff
@@ -1,12 +1,131 @@
...
+                    contentStyle={{ background: '#631111', color: '#fff' }}
+                    contentArrowStyle={{ borderRight: '7px solid  #631111' }}
+                    iconStyle={{ background: '#631111', color: '#fff' }}
+                >
+                    <h3 className={styles.TimeLineHeading}>{rover.name}</h3>
+                    <img className={styles.RoverImage} src={rover.imageUrl} alt="rover" />
+                    <p className={styles.RoverInformation}>
+                        {rover.information}
+                    </p>
+                </VerticalTimelineElement>
```
Comment (MichaelWalker) [inline src/TimelineID/TimelineId.tsx:47]: It looks like there is quite a lot in common between these elements, so I'd be tempted to make our own component to handle some of this stuff.

```tsx
export const TimelineElement: FunctionComponent<TimeLineElementProps> = () => {
    return (
        <VerticalTimelineElement
               ....
    );
} 
```

File: src/TimelineID/TimelineId.tsx [hunk 3]
```diff
@@ -1,12 +1,131 @@
...
+                    date="April 2013"
+                    contentStyle={{ background: '#631111', color: '#fff' }}
+                    contentArrowStyle={{ borderRight: '7px solid  #631111' }}
+                    iconStyle={{ background: '#631111', color: '#631111' }}
+                >
+                    <h3 className={styles.TimeLineHeading}>{rover.name}</h3>
+                    <img className={styles.RoverImage} src={rover.imageUrl} alt="rover" />
+                    <p className={styles.RoverInformation}>
+                        {rover.information}
+                    </p>
```
Comment (MichaelWalker) [inline src/TimelineID/TimelineId.tsx:61]: so at the moment it just repeats the same data multiple times.
When its used for real what data are we expecting it to display?

File: src/TimelineID/TimelineId.tsx [hunk 4]
```diff
@@ -1,12 +1,131 @@
...
+                </VerticalTimelineElement>
+            </VerticalTimeline>
+            <Link to="timeline/:roverName" className={styles.ButtonToRoverImage}>{rover.name} Image explorer</Link>
+        </section> //change the line to when image viewer is complete
+    )
+};
+
+export function getRover(roverName: string): Rover | undefined {
+    console.log(roverName);
```
Comment (MichaelWalker) [inline src/TimelineID/TimelineId.tsx:110]: lost `console.log` :)

**Review by MichaelWalker — CHANGES_REQUESTED** [review]
Looks like CI build is failing, and a few little points, but overall looking really good.

File: src/TimelineID/TimelineID.test.tsx [hunk 5]
```diff
@@ -1,10 +1,18 @@
...
-    const linkElement = getByText(/The current Timeline Id is:/i);
-    expect(linkElement).toBeInTheDocument();
+test('renders Rover Timeline', () => {
+    const queries = render(<MemoryRouter initialEntries={["/timeline/perserverance"]}>
+        
+            <Route path="/timeline/:roverName">
+                <TimelineId />
+            </Route>
+    </MemoryRouter>);
+    mockAllIsIntersecting(true);
```
Comment (MichaelWalker) [inline src/TimelineID/TimelineID.test.tsx:14]: does this just make sure that everything is 'onScreen'?

**Review by MichaelWalker — APPROVED** [review]
Nice - looks great.
Can't wait to see it once its merged in

File: src/TimelineID/TimelineID.test.tsx (hunk 5 above)
Comment (Codringtoncoding) [reply src/TimelineID/TimelineID.test.tsx:14]: Yes I had to mock a test because something was clashing with the intersection scope property in react

**Pull request closed without merging**